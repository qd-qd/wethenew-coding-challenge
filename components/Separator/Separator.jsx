import PropTypes from 'prop-types';
import styles from './Separator.module.css';

const Separator = ({ className }) => (
  <hr className={`${styles.separator} ${className}`} />
);

Separator.defaultProps = {
  className: '',
};

Separator.propTypes = {
  className: PropTypes.string,
};

export default Separator;
