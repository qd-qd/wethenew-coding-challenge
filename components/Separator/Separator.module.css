.separator {
  color: var(--theme-color-grey-800);
  opacity: 0.4;
}
