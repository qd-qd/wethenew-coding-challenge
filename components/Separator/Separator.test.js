import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Separator from './Separator';

describe('Separator', () => {
  test('displays greeting', () => {
    const { container } = render(<Separator />);
    expect(container.firstChild.classList.contains('separator')).toBe(true);
  });

  test('displays custom className', () => {
    const className = 'fake_class';
    const { container } = render(<Separator className={className} />);
    expect(container.firstChild.classList.contains('separator')).toBe(true);
    expect(container.firstChild.classList.contains(className)).toBe(true);
  });

  test("doesn't break snapshot", () => {
    const { container } = render(<Separator />);

    expect(container).toMatchSnapshot();
  });
});
