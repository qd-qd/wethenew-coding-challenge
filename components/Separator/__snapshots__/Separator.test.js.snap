// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator doesn't break snapshot 1`] = `
<div>
  <hr
    class="separator "
  />
</div>
`;
